### PR TITLE
Installer OUTFD fixes and emulator support

### LINKED ISSUE #813: Installation on Genymotion devices using drag n drop fails
**Issue description:**

Installation on Genymotion devices using drag n drop fails

**Reproduction Steps:**

1. Get the latest x86/x86_64 archive of Opengapps 
2. Drag n drop this archive on a running Genymotion device (desktop or int the Cloud)
3. Genymotion will detect this as an opengapps archive and ask if you want to continue the installation
4. Click yes
5. Genymotion displays an error in the installation process
6. Device is not accessible anymore using adb shell

**Expected behavior:**

Package should be installed correctly and device should indicate the user it needs to be restarted

**Observed behavior:**

Package installation (installer.sh) returns an error and device is now in an unusable state

(Patch incoming from Genymotion)

### PR DESCRIPTION
**Pull request by osm0sis**
installer: fix OUTFD use when redirected to a log, cleanups:
- append to OUTFD to avoid truncating when redirected to a log file
- use ui_print in backend to simplify error output

installer: support emulator drag-and-drop installs:
- Fixes #813, closes #814

### CODE REVIEW

File: scripts/inc.installer.sh [hunk 1]
```diff
@@ -1106,12 +1107,21 @@ ui_print " "
 # For reference, check https://github.com/osm0sis/AnyKernel3/blob/master/META-INF/com/google/android/update-binary
 ui_print "- Mounting partitions"
 set_progress 0.01
+
+ps | grep zygote | grep -v grep >/dev/null && BOOTMODE=true || BOOTMODE=false
+$BOOTMODE || ps -A 2>/dev/null | grep zygote | grep -v grep >/dev/null && BOOTMODE=true
```
Comment (JulienBolard) [inline scripts/inc.installer.sh]: Maybe we can use getprop here instead using
[init.svc.zygote]: [running]

Comment (osm0sis) [reply scripts/inc.installer.sh]: This is the established script for $BOOTMODE, established by Magisk, and is known working in all cases, so I'd rather not mess with it.

**Comment by osm0sis** [comment]
@JulienBolard can you confirm an Open GApps zip built with this PR merged works?

**Comment by JulienBolard** [comment]
Current builds on 'master' branch for api 28 and 29 get installed successfully after drag n drop, BUT I have a boot loop when devices rebooted.
I need to investigate why https://pastebin.com/60pnsUbM

**Comment by mfonville** [comment]
@JulienBolard the error is related to a duplicate instance of the package-manager app. Maybe there is both an AOSP/customized/OpenGApps apk that serves that function installed?

**Comment by JulienBolard** [comment]
> @JulienBolard the error is related to a duplicate instance of the package-manager app. Maybe there is both an AOSP/customized/OpenGApps apk that serves that function installed?

It was the opposite: No package Installer was present after my reboot. It seems to come from the .zip I built. 
It works without any problem with march 7 nightly.

**Pull request merged**